#164035303 Delete Office Endpoint

odaridavid opened the pull request:
### What does this PR do?
- Adds Delete Office Route with persistence to a database
- Configures env file to pick DB depending on env
- Adds tests for delete office endpoint

### Description of Task To Be Completed?
Have the DELETE method working on `/offices/office_id` endpoint working for both success and error cases
### How can this be manually tested?
- Clone the repo and checkout to bg-office-persistence-164035303
- Ensure db server is installed and in the env paste URIs that point to your db
- start the flask app using flask run and run on an api testing client to observe persistence with db
### Relevant Pivotal Tracker Stories?
[#164035303](https://www.pivotaltracker.com/n/projects/2241897)
### Screenshots
##### Deleting Office Process
1. Create Party
![screenshot from 2019-02-19 11-52-27](https://user-images.githubusercontent.com/20339228/53002293-6f82c000-343d-11e9-9914-f51723d229cb.png)
2. Delete Party
![screenshot from 2019-02-19 11-53-49](https://user-images.githubusercontent.com/20339228/53002311-7dd0dc00-343d-11e9-92e7-ee78e1edfa84.png)
3. Confirm List Of Party that it doesnt exist
![screenshot from 2019-02-19 11-53-59](https://user-images.githubusercontent.com/20339228/53002338-8de8bb80-343d-11e9-8695-b3da31e01060.png)

odaridavid committed:
[Bug #164035303]Adds Edge Case to get data

odaridavid committed:
[Finishes #164035303]Add Delete Office Endpoint

- Adds Delete Office route in view and functionality in  model
- configure env in config
- Add tests for office model and view for deleting

odaridavid committed:
[Finishes #164035303]Refactor Travis for db

odaridavid committed:
[Finishes #164035303]Refactor Result for offices to come as dicts in list

- default config set to development

meshack-mbuvi commented on tests/v2tests/test_office_view_db.py:
```diff
@@ -121,7 +121,26 @@ def test_specific_item_success(self):
         self.assertIn('Government Steward', response.json['data'][0]['office_name'])
 
-    def test_specific_item_fail(self):
+    def test_specific_office_unsuccessful(self):
```
This could be improved by naming it **test_get_non_existing_office**

odaridavid committed:
[Finishes #164035303] Code Cleanup and bug fixes

- Removes unnecessary comments
- Edit tests methods for clarity
- Removed badges from README for codacy and code climate maintainability
- Fix Edit bug where an office could be updated even if another similar one
exists

odaridavid committed:
[Finishes #164035303] Code Cleanup

- Rename variables using built in variable names

odaridavid committed:
[Finishes #164035303] Added Check if nothing exists to edit

odaridavid committed:
[Finishes #164035303] Fix failing test for update

odaridavid committed:
[Finishes #164035303] Fix failing test for update

odaridavid committed:
[Finishes #164035303] Fixes failing test for update

odaridavid committed:
[Finishes #164035303] Fixes failing test for update ,typo issue

odaridavid committed:
[Finishes #164035303] Add .env with no db uri

odaridavid committed:
Merge remote-tracking branch 'origin/bg-office-persistence-164035303' into bg-office-persistence-164035303

odaridavid committed:
[Finishes #164035303]Remove unnecesary comments and drop table function
